python/pip_install/pip_repository.bzl

```diff
@@ -61,6 +64,27 @@ def _resolve_python_interpreter(rctx):
             fail("python interpreter `{}` not found in PATH".format(python_interpreter))
     return python_interpreter
 
+def _maybe_set_xcode_location_cflags(rctx, environment):
+    """Patch environment with CFLAGS of xcode sdk location.
+
+    Figure out if this interpreter target comes from rules_python, and patch the xcode sdk location if so.
+    Pip won't be able to compile c extensions from sdists with the pre built python distributions from indygreg
+    otherwise. See https://github.com/indygreg/python-build-standalone/issues/103
+    """
+    if (
+        rctx.os.name.lower().startswith("mac os") and
+        rctx.attr.python_interpreter_target != None and
+        rctx.path(Label("@{}//:{}".format(rctx.attr.python_interpreter_target.workspace_name, STANDALONE_INTERPRETER_FILENAME)))
+    ):
+        xcode_sdk_location = rctx.execute(["xcode-select", "-p"])
+        if xcode_sdk_location.return_code == 0:
+            xcode_root = xcode_sdk_location.stdout.strip()
+            if "commandlinetools" not in xcode_root.lower():
+                # This is a full xcode installation somewhere like /Applications/Xcode13.0.app/Contents/Developer
+                # so we need to change the path to to the macos specific tools.
+                xcode_root = "{}/Platforms/MacOSX.platform/Developer".format(xcode_root)
+            environment[CFLAGS] = "-isysroot {}/SDKs/MacOSX.sdk".format(xcode_root)
```

Comment:
Yeah it will, good catch. I believe I was checking for an existing CFLAGS before but must have lost it in some refactoring. My idea was to not overwrite it if CFLAGS was already set in the environment with the assumption that the user knows what they want if they are specifically setting CFLAGS.